src/meta/src/manager/env.rs

```diff
@@ -416,11 +417,17 @@ impl MetaSrvEnv {
                     .await?
                     .map(|c| c.cluster_id.to_string().into())
                     .unwrap();
+
+                let mut system_params = init_system_params;
+                // For new clusters, the name of the object store needs to be prefixed according to the object id.
+                // For old clusters, the prefix is ​​not divided for the sake of compatibility.
+                system_params.use_new_object_prefix_strategy =
+                    Some(is_sql_backend_cluster_first_launch);
```

Comment:
Did we forget to set `use_new_object_prefix_strategy` when etcd backend is used in L377?

Reply:
No, for etcd backend, aka SQL backend, we will use the origin variable cluster_first_launch as the `use_new_object_prefix_strategy`, see L71 in `system_param/mod.rs`.
```
 params.use_new_object_prefix_strategy = Some(cluster_first_launch);
```

Reply:
This is incorrect. Consider the following case:
1. Cluster is created with the new image after this PR.
2. `cluster_first_launch` is true on meta pod starts up for the first time and `use_new_object_prefix_strategy` is persisted to be true in etcd.
3. meta pod restarts
4. `cluster_first_launch` is false this time, but we will still hit L71 in `system_param/mod.rs`, resulting in `use_new_object_prefix_strategy` mistakenly changed to false.

This is all because we change the system params value after `merge_params` in `SystemParamsManager::new`. I think we should following the semantic of `init_params` exactly and provide the initial value of `use_new_object_prefix_strategy` in `init_params` here outside of `SystemParamsManager::new`